src/status.ts

```diff
@@ -23,123 +30,124 @@ const STARTED_SERVER_OPTIONS = [
   { label: "Ruby LSP: Restart", description: Command.Restart },
 ];
 
-export class StatusItem {
-  private context: vscode.ExtensionContext;
-  private ruby: Ruby;
-  private selector: vscode.DocumentSelector;
-  private serverStatus: vscode.LanguageStatusItem;
-  private yjitStatus: vscode.LanguageStatusItem;
-  private rubyVersionStatus: vscode.LanguageStatusItem;
-  private experimentalFeaturesStatus: vscode.LanguageStatusItem;
-
-  constructor(context: vscode.ExtensionContext, ruby: Ruby) {
-    this.context = context;
-    this.ruby = ruby;
-    this.selector = {
+export interface Client {
```

Comment:
It would be great if we could rename this to `ClientInterface` (or similar) and then used it in `client.ts` as `class Client implements ClientInterface`. That would ensure that the interface cannot go out of sync and makes the contract more explicit.

Reply:
Great idea.